src/idpi/mch_model_data.py

```diff
@@ -0,0 +1,82 @@
+"""Get Meteoswiss model data from FDB or polytope."""
+
+# Standard library
+import logging
+import os
+
+# Third-party
+import xarray as xr
+
+# Local
+from . import data_source, grib_decoder, mars
+
+logger = logging.getLogger(__name__)
+
+
+def get_from_fdb(request: mars.Request) -> dict[str, xr.DataArray]:
```

Comment:
Could the `request` also be of type `str | tuple | dict` as seen in other places in the code?

Reply:
Here it's better to ensure a full request because all keys need to be specified when making requests to FDB or polytope. In the rest of the code, mainly in the tests, the request is done against files and a request template is used to complete the request for FDB but this approach is a bit complicated for this API.